### DISCUSSION TITLE
Running page code from unsafeWindow without privileges

### DISCUSSION THREAD

**Discussion by Glandos** [violentmonkey/violentmonkey#1267] [Q&A] [answered]
I am trying to read context menus created with [jQuery-contextmenu](https://github.com/swisnl/jQuery-contextMenu). These are created with `$.contextMenu()` function.
One of my idea was to redefine this function using `unsafeWindow.$.contextMenu = […]` and intercept options, but then, I realize that some options for that menus are functions that I must call (with jQuery elements as arguments).
However, this seems dangerous to me: those functions are defined in the web page, and can be injected by anyone, to be executed in my script context.

Is there a way to run those functions within page context, and getting the result (that are plain object) somehow?

**Comment by gera2ld** [comment, marked as answer]
Those functions are defined in the web page so they don't have access to any `GM_*` APIs. Your script is executed roughly in a closure, so it should be safe as long as you don't pass anything defined in the closure to the unsafe functions by arguments or `this`.

**Reply by Glandos** [reply]
Thanks. I was afraid that the called function could escape some security like Same Site when doing XHR, but I guess it can't.
